File: Leetcode/75 - Sort Colors/solution.py
def sortColors(self, nums: List[int]) -> None:
        low: int = 0
        mid: int = 0
        high: int = len(nums) - 1

        while mid <= high:
            value = nums[mid]
            if value == 0:
                nums[low], nums[mid] = nums[mid], nums[low]
                low += 1
                mid += 1
            elif value == 1:
                mid += 1
            else:
                nums[mid], nums[high] = nums[high], nums[mid]
                high -= 1

File: Leetcode/75 - Sort Colors/solution.ts
function sortColors(nums: number[]): void {
    let low: number = 0;
    let mid: number = 0;
    let high: number = nums.length - 1;

    while (mid <= high) {
        if (nums[mid] === 0) {
            [nums[low], nums[mid]] = [nums[mid], nums[low]];
            low++;
            mid++;
        } else if (nums[mid] === 2) {
            [nums[mid], nums[high]] = [nums[high], nums[mid]];
            high--;
        } else {
            mid++;
        }
    }
};